core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -145,6 +146,7 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
   private val members = new mutable.HashMap[String, MemberMetadata]
   private val offsets = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
   private val pendingOffsetCommits = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
+  private val deletedPartitions = new mutable.HashSet[TopicPartition]
```

Comment:
@hachikuji Thanks for the alternate logic. new logic is much simpler with minimum changes.
required changes are done.

One query: Upon partition deletion, For commit pending offsets, we are appending to log, but we are not adding to cache. Will this cause any issue during immediate failover/or any other scenario?